src/providers/r2Provider.ts

```diff
@@ -85,9 +86,33 @@ export class R2Provider implements Provider {
     path: string,
     options?: ReadDirectoryOptions
   ): Promise<ReadDirectoryResult | undefined> {
+    if (path in CACHED_DIRECTORIES) {
+      // @ts-expect-error dates may not be parsed at this point, we take care
+      //  of it below
+      const result: ReadDirectoryResult = CACHED_DIRECTORIES[path];
```

Comment:
Let's make a temporary type then where the dates are not parsed yet. And then the actual result type receives a generic of the previous type.